Sorry I was late reply....  Now I tried to test following environments.  As a result, I think it's OK to support testing on Ubuntu16.04 only.  In addition, please add following command for testing.

```
sudo apt install cmake
```

Here are operation logs:

* MacOS High Sierra 10.13.2
    * An error occurred while running 'make'

        ```
        error: invalid argument '-std=c++11' not allowed with 'C/ObjC'
        make[2]: *** [CMakeFiles/unittests.dir/Users/u4r/workspace/retro-typing/typing.c.o] Error 1
        make[1]: *** [CMakeFiles/unittests.dir/all] Error 2
        make: *** [all] Error 2
        ```
    * cmake --version

        ```
        cmake version 3.10.2
        ```
* Ubuntu16.04

    * cmake --version
        ```
        cmake version 3.5.1
        ```
    * It worked!  However, it's better to add  **sudo apt install cmake**  before 'cmake' in CUSTOMIZE.md

## Ubuntu16.04

* It's better to describe required packages 'cmake'.  Following error occurred in running 'cmake':

    ```
    The program 'cmake' is currently not installed. You can install it by typing:
    sudo apt install cmake
    ```

* run 'cmake .' after installed 'cmake':

    ```
    CMake Warning (dev) in CMakeLists.txt:
      No cmake_minimum_required command is present.  A line of code such as
     
    cmake_minimum_required(VERSION 3.5)
    
    should be added at the top of the file.  The version specified may be lower
    if you wish to support older CMake versions for this project.  For more
    information run "cmake --help-policy CMP0000".
    This warning is for project developers.  Use -Wno-dev to suppress it.

    -- Configuring done
    CMake Warning (dev) at CMakeLists.txt:42 (add_executable):
      Policy CMP0003 should be set before this line.  Add code such as

        if(COMMAND cmake_policy)
          cmake_policy(SET CMP0003 NEW)
        endif(COMMAND cmake_policy)

      as early as possible but after the most recent call to
      cmake_minimum_required or cmake_policy(VERSION).  This warning appears
      because target "unittests" links to some libraries for which the linker
      must search:
      
        ncursesw, tinfo

      and other libraries with known full path:

        /home/yushiro/retro-typing/unittests/googletest-build/googlemock/gtest/libgtest_main.a

      CMake is adding directories in the second list to the linker search path in
      case they are needed to find libraries from the first list (for backwards
      compatibility with CMake 2.4).  Set policy CMP0003 to OLD or NEW to enable
      or disable this behavior explicitly.  Run "cmake --help-policy CMP0003" for
      more information.
      This warning is for project developers.  Use -Wno-dev to suppress it.
      
    -- Generating done
    -- Build files have been written to: /home/yushiro/retro-typing/unittests
    ```
* Run 'make'

    ```
    ...(snip)...
    [ 88%] Building CXX object CMakeFiles/unittests.dir/typing_test.o
    [100%] Linking CXX executable unittests
    [100%] Built target unittests
    ```
* Run './unittests'

    ```
    Running main() from gtest_main.cc
    [==========] Running 3 tests from 1 test case.
    [----------] Global test environment set-up.
    [----------] 3 tests from TypingTest
    [ RUN      ] TypingTest.ExitFromRetroTyping
    Running main() from gtest_main.cc
    [       OK ] TypingTest.ExitFromRetroTyping (4 ms)
    [ RUN      ] TypingTest.CToI
    [       OK ] TypingTest.CToI (0 ms)
    [ RUN      ] TypingTest.GetComboMsgYeah
    [       OK ] TypingTest.GetComboMsgYeah (0 ms)
    [----------] 3 tests from TypingTest (4 ms total)
    
    [----------] Global test environment tear-down
    [==========] 3 tests from 1 test case ran. (4 ms total)
    [  PASSED  ] 3 tests.
    ```

--- a/CUSTOMIZE.md
+++ b/CUSTOMIZE.md
@@ -46,7 +46,7 @@ rm ranking/*.log
 # How to test
 
 This project has unit tests using googletest testing framework.
-You can build and execute unit tests as follows.
+You can build and execute unit tests as follows. **Note that currently not to support MacOS**.
 
 **Requirements**: cmake version 3.1 or later.
 

--- a/CUSTOMIZE.md
+++ b/CUSTOMIZE.md
@@ -57,10 +57,8 @@ sudo apt install cmake
 ```
 
 ```
-cd unittests/
-mkdir build
-cd build
-cmake ..
+cd unittests
+cmake .
 make
 ./unittests
 ```

--- a/CUSTOMIZE.md
+++ b/CUSTOMIZE.md
@@ -48,9 +48,19 @@ rm ranking/*.log
 This project has unit tests using googletest testing framework.
 You can build and execute unit tests as follows.
 
+**Requirements**: cmake version 3.1 or later.
+
+Install cmake before you build unit tests.
+
+```
+sudo apt install cmake
+```
+
 ```
 cd unittests/
-cmake .
+mkdir build
+cd build
+cmake ..
 make
 ./unittests
 ```

--- a/unittests/CMakeLists.txt
+++ b/unittests/CMakeLists.txt
@@ -1,3 +1,13 @@
+cmake_minimum_required(VERSION 3.1)
+project(retro-typing)
+enable_language(CXX)
+set(CMAKE_CXX_STANDARD 11) # C++11
+set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wall")
+
+if(COMMAND cmake_policy)
+  cmake_policy(SET CMP0003 NEW)
+endif(COMMAND cmake_policy)
+
 # Download and unpack googletest at configure time
 configure_file(CMakeLists.txt.in googletest-download/CMakeLists.txt)
 execute_process(COMMAND ${CMAKE_COMMAND} -G "${CMAKE_GENERATOR}" .
@@ -34,9 +44,6 @@ endif()
 find_package(PkgConfig)
 pkg_check_modules(NCURSESW REQUIRED ncursesw)
 include_directories(${NCURSESW_INCLUDE_DIRS} ../)
-
-# Unit tests use c++11
-add_definitions("-Wall -std=c++11")
 
 # Now simply link against gtest or gtest_main as needed. Eg
 add_executable(unittests ../typing.c ../logging.c ../display.c typing_test.cpp)

